cloud/cluster/cluster.go

```diff
@@ -0,0 +1,65 @@
+package cluster
```

Comment:
Yea i'm a bit confused why we are refactoring this out of scheduler?  What other code is going to use it?